apps/test/unit/code-studio/pd/form_components/button_list_test.js

```diff
@@ -0,0 +1,221 @@
+
+describe("ButtonList", () => {
+  describe("With type: radio", () => {
+    let radioList;
+    let onChangeCallback;
+
+    before(() => {
+      onChangeCallback = sinon.spy();
+
+      radioList = shallow(
+        <ButtonList
+          type="radio"
+          label="What is your favorite pet?"
+          groupName="favoritePet"
+          answers={[
+            "Cat",
+            "Dog"
+          ]}
+          onChange={onChangeCallback}
+        />
+      );
+    });
+
+    it("Renders radio buttons", () => {
+      expect(radioList).to.containMatchingElement(
+        <FormGroup id="favoritePet" controlId="favoritePet">
+          <ControlLabel>
+            What is your favorite pet?
+          </ControlLabel>
+          <FormGroup>
+            <Radio value="Cat" label="Cat" name="favoritePet">
+              Cat
+            </Radio>
+            <Radio value="Dog" label="Dog" name="favoritePet">
+              Dog
+            </Radio>
+          </FormGroup>
+          <br />
+        </FormGroup>
+      );
+    });
+
+    it("Calls the onChange callback with a single result when one is selected", () => {
+      radioList.find("Radio[value='Cat']").simulate("change", {target: {value: "Cat"}});
+      expect(onChangeCallback).to.have.been.calledOnce;
+      expect(onChangeCallback).to.have.been.calledWith({favoritePet: "Cat"});
+    });
+  });
+
+  describe("With type: check", () => {
+    let checkboxList;
+    let onChangeCallback;
+
+    before(() => {
+      onChangeCallback = sinon.spy();
+
+      checkboxList = shallow(
+        <ButtonList
+          type="check"
+          label="What is your favorite pet?"
+          groupName="favoritePet"
+          answers={[
+            "Cat",
+            "Dog"
+          ]}
+          onChange={onChangeCallback}
+        />
+      );
+    });
+
+    it("Renders checkboxes", () => {
+      expect(checkboxList).to.containMatchingElement(
+        <FormGroup id="favoritePet" controlId="favoritePet">
+          <ControlLabel>
+            What is your favorite pet?
+          </ControlLabel>
+          <FormGroup>
+            <Checkbox value="Cat" label="Cat" name="favoritePet">
+              Cat
+            </Checkbox>
+            <Checkbox value="Dog" label="Dog" name="favoritePet">
+              Dog
+            </Checkbox>
+          </FormGroup>
+          <br/>
+        </FormGroup>
+      );
+    });
+
+    it("Calls the onChange callback with a list of all checked when one is changed", () => {
+      // Select "Cat" from [], resulting in ["Cat"]
+      checkboxList.find("Checkbox[value='Cat']").simulate("change", {target: {value: "Cat", checked: true}});
+      expect(onChangeCallback).to.have.been.calledOnce;
+      expect(onChangeCallback).to.have.been.calledWith({favoritePet: ["Cat"]});
+
+      // Unselect "Cat" from ["Cat", "Dog"], resulting in ["Dog"]
+      checkboxList.setProps({...checkboxList.props.merge, selectedItems: ["Cat", "Dog"]});
```

Comment:
I just noticed I have this extraneous `merge` here (likely from Ruby-fingers), which is not a prop, so this is spreading `undefined` which means clearly the original props aren't needed here. I'll remove that part and only set `selectedItems`